File: cpe49題/11349 Symmetric Matrix.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
	int t,caseNum=1;
	cin>>t;
	
	while(t--){
		string s;
		bool isSymmetric = true;
		int arrayWide,array[100][100]={0};
		cin>>s>>s;
		cin>>arrayWide;
		
		for(int i=0;i<arrayWide;i++){
			for(int j=0;j<arrayWide;j++){
				int num;
				cin>>num;
				
				array[i][j]=num;
			}
		}
		
		for(int i=0;i<arrayWide;i++){
			for(int j=0;j<arrayWide;j++){
				if(array[i][j]<0){
					isSymmetric=false;
				}
				if(array[i][j]!=array[arrayWide-1-i][arrayWide-1-j]){
					isSymmetric=false;
				}
			}
		}
		
		if(isSymmetric){
			cout<<"Test #"<<caseNum++<<": Symmetric."<<endl;
		}else{
			cout<<"Test #"<<caseNum++<<": Non-symmetric."<<endl;	
		}
		
	}
}

File: cpe49題/11349 Symmetric Matrix.py
caseNum=1

while True:
	try:
		n=int(input())
		
		for _ in range(n):
			numList=[]
			s=input().strip()
			
			for _ in range(int(s[-1])):
				tmp=list(map(int,input().strip().split()))
				numList+=tmp
			
			isSym=True
			for i in range(len(numList)):
				left=numList[i]
				right=numList[len(numList)-1-i]
				
				if left!=right or left<0 or right<0:
					isSym=False				
					break
					
			if isSym:					
				print(f"Test #{caseNum}: Symmetric.")
			else:
				print(f"Test #{caseNum}: Non-symmetric.")
			caseNum+=1
			
	except EOFError:
		break
